### ISSUE TITLE
Custom Dataset

### ISSUE THREAD

**Issue by Tonks684** [wvangansbeke/Unsupervised-Semantic-Segmentation#3]
Thank you very much, this is very exciting work. I was wondering if you could share any guidance on how to customize the code for alternative datasets?

**Comment by wvangansbeke** [comment]
Hi @Tonks684,
Thank you for your interest! 

In the first step you will need to generate mask proposals. Depending on the dataset of interest, you can adopt a mid-level visual prior (e.g. saliency, optical flow, etc.) to obtain these masks. In the second step, you apply the loss function from in the paper (Eq. 3) using the object masks. Have a look [here](https://github.com/wvangansbeke/Unsupervised-Semantic-Segmentation/blob/main/pretrain/modules/moco/builder.py#L117) for more details about the loss. 

Regarding the code. You will have to write a dataloader for your custom dataset that contains the images and their corresponding object mask proposals. Definitely take a look [here](https://github.com/wvangansbeke/Unsupervised-Semantic-Segmentation/blob/main/pretrain/data/dataloaders/pascal_voc.py) and check how we use the saliency masks. 

Btw, we are planning to update parts of the paper and make some additions to this repository in the near future. 

Hope this is helpful.

**Comment by wvangansbeke** [comment]
If you have any further issues, let us know.

**Comment by Tonks684** [comment]
Thank you for your explanation. 

I'm trying to run the DeepUSPS to generator the initial saliency masks. The requirement.txt provided by https://tinyurl.com/wtlhgo3 is causing me to have a lot of incompatible libraries. Are you able to share the requirements.txt you used or the changes you made? I imagine their file is out of date perhaps. Thank you in advance!

Sam.
